uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/providers/IdentityProviderEndpointsAliasMockMvcTests.java

```diff
@@ -327,229 +373,351 @@ protected UpdateBase(final boolean aliasFeatureEnabled) {
-                assertRelyingPartySecretIsRedacted(updatedOriginalIdp);
-            }
+                @Test
+                void shouldReject_ReferencedZoneDoesNotExist() throws Exception {
+                    final IdentityZone zone = IdentityZone.getUaa();
+                    final IdentityProvider<?> existingIdp = createIdp(
+                            zone,
+                            buildUaaIdpWithAliasProperties(zone.getId(), null, null)
+                    );
 
-            @Test
-            void shouldAccept_ReferencedIdpNotExisting_ShouldCreateNewAliasIdp_UaaToCustomZone() throws Exception {
-                shouldAccept_ReferencedIdpNotExisting_ShouldCreateNewAliasIdp(IdentityZone.getUaa(), customZone);
-            }
+                    existingIdp.setAliasZid(UUID.randomUUID().toString()); // non-existing zone
 
-            @Test
-            void shouldAccept_ReferencedIdpNotExisting_ShouldCreateNewAliasIdp_CustomToUaaZone() throws Exception {
-                shouldAccept_ReferencedIdpNotExisting_ShouldCreateNewAliasIdp(customZone, IdentityZone.getUaa());
-            }
+                    shouldRejectUpdate(zone, existingIdp, HttpStatus.UNPROCESSABLE_ENTITY);
+                }
 
-            private void shouldAccept_ReferencedIdpNotExisting_ShouldCreateNewAliasIdp(final IdentityZone zone1, final IdentityZone zone2) throws Exception {
-                final IdentityProvider<?> idp = createIdpWithAlias(zone1, zone2);
+                @Test
+                void shouldReject_AliasNotSupportedForIdpType_UaaToCustomZone() throws Exception {
+                    shouldReject_AliasNotSupportedForIdpType(IdentityZone.getUaa(), customZone);
+                }
 
-                // delete the alias IdP directly in the DB -> after that, there is a dangling reference
-                deleteIdpViaDb(idp.getOriginKey(), zone2.getId());
+                @Test
+                void shouldReject_AliasNotSupportedForIdpType_CustomZone() throws Exception {
+                    shouldReject_AliasNotSupportedForIdpType(customZone, IdentityZone.getUaa());
+                }
 
-                // update some other property on the original IdP
-                idp.setName("some-new-name");
-                final IdentityProvider<?> updatedIdp = updateIdp(zone1, idp);
-                assertThat(updatedIdp.getAliasId()).isNotBlank().isNotEqualTo(idp.getAliasId());
-                assertThat(updatedIdp.getAliasZid()).isNotBlank().isEqualTo(idp.getAliasZid());
+                private void shouldReject_AliasNotSupportedForIdpType(final IdentityZone zone1, final IdentityZone zone2) throws Exception {
+                    final IdentityProvider<?> uaaIdp = buildUaaIdpWithAliasProperties(zone1.getId(), null, null);
+                    final IdentityProvider<?> createdProvider = createIdp(zone1, uaaIdp);
+                    assertThat(createdProvider.getAliasZid()).isBlank();
```

Comment:
Looks good. Leaving comments to track my progress as this is challenging to read through.